File: 프로그래머스/전화번호목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    sort(phone_book.begin(), phone_book.end());
    for(int i=0; i<phone_book.size()-1; i++){
        answer=false;
        for(int j=0; j<phone_book[i].length(); j++)
            if(phone_book[i][j]!=phone_book[i+1][j]) answer=true;

        if(!answer) return answer;
    }

    return answer;
}

File: 프로그래머스/전화번호목록.java
import java.util.Arrays;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        for(int i=0, len=phone_book.length; i<len-1; i++){
            int len1 = phone_book[i].length();
            int len2 = phone_book[i+1].length();
            if(len2<=len1) continue;
            if(phone_book[i+1].substring(0, len1).equals(phone_book[i])) return false; 
            
        }
        
        return true;
    }
}